docs/src/content/hardhat-runner/docs/guides/typescript.md

```diff
@@ -10,188 +10,131 @@ Hardhat will automatically enable its TypeScript support if your config file end
+You can easily turn a JavaScript Hardhat config file into a TypeScript one. Let's see how this is done starting with a fresh Hardhat project.
 
-## Writing tests and scripts in TypeScript
+Open your terminal, go to an empty folder, run `npx hardhat`, and go through the steps to create a JavaScript project. When you're done your project directory should look something like this:
 
-To write your smart contract tests and scripts you'll most likely need access to an Ethereum library to interact with your smart contracts. This will probably be one of [hardhat-ethers](https://github.com/nomiclabs/hardhat/tree/master/packages/hardhat-ethers) or [hardhat-web3](https://github.com/nomiclabs/hardhat/tree/master/packages/hardhat-web3), all of which inject instances into the [Hardhat Runtime Environment].
+```
+$ ls -l
+total 1200
+drwxr-xr-x    3 pato  wheel      96 Oct 20 12:50 contracts/
+-rw-r--r--    1 pato  wheel     567 Oct 20 12:50 hardhat.config.js
+drwxr-xr-x  434 pato  wheel   13888 Oct 20 12:52 node_modules/
+-rw-r--r--    1 pato  wheel  604835 Oct 20 12:52 package-lock.json
+-rw-r--r--    1 pato  wheel     460 Oct 20 12:52 package.json
+drwxr-xr-x    3 pato  wheel      96 Oct 20 12:50 scripts/
+drwxr-xr-x    3 pato  wheel      96 Oct 20 12:50 test/
+```
 
-When using JavaScript, all the properties in the HRE are injected into the global scope, and are also available by getting the HRE explicitly. When using TypeScript nothing will be available in the global scope and you will need to import everything explicitly.
+Then, you should follow the steps mentioned in the [Installing dependencies](#installing-dependencies) section above.
 
-An example for tests:
+Now, we are going to rename the config file from `hardhat.config.js` to `hardhat.config.ts`, just run:
 
-```typescript
-import { ethers } from "hardhat";
-import { Signer } from "ethers";
+```
+mv hardhat.config.js hardhat.config.ts
+```
 
-describe("Token", function () {
-  let accounts: Signer[];
+We need to a single change to your config for it to work with TypeScript: plugins must be loaded with `import` instead of `require`.
 
-  beforeEach(async function () {
-    accounts = await ethers.getSigners();
-  });
+By using TypeScript, you can also type your configuration, which will save you from typos and other mistakes.
 
-  it("should do something right", async function () {
-    // Do something with the accounts
-  });
-});
-```
+For example, the sample project's config turns from this:
 
-An example for scripts:
+<<< @/../packages/hardhat-core/sample-projects/javascript/hardhat.config.js{1}
 
-```typescript
-import { run, ethers } from "hardhat";
+into this:
 
-async function main() {
-  await run("compile");
+<<< @/../packages/hardhat-core/sample-projects/typescript/hardhat.config.ts{1,2,4-6}
 
-  const accounts = await ethers.getSigners();
+Finally, you need to create a [`tsconfig.json`](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html) file. Here's our recommended one:
```

Comment:
Shouldn't we use `es2020`? Or does `es2017` support native bigints?

Reply:
We can't use it until node 12 gets EOL'd. ES2020 doesn't downcompiler `{...this}`, and node 12 doesn't support it.

Reply:
I see, thanks for the explanation.